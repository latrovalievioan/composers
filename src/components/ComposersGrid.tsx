import { Composer } from '../types';
import { useAppSelector } from '../hooks';
import { ComposerCard } from './ComposerCard';
import './ComposersGrid.css';

export const ComposersGrid = () => {
  const composerList = useAppSelector((state) => state.composerList);
  return (
    <div className="composer-grid">
      {composerList.map((composer: Composer) => {
        return <ComposerCard key={composer.id} composer={composer} />;
      })}
    </div>
  );
};
