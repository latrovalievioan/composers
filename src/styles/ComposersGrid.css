.composer-grid {
    display:grid;
    grid-template-columns: repeat(auto-fit, 23rem);
    justify-content: center;
    gap: 2rem;
    width: 100vw;
}
