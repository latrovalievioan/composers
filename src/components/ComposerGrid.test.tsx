import React from 'react';
import renderer from 'react-test-renderer';
import { ComposersGrid } from './ComposersGrid';
import { BrowserRouter } from 'react-router-dom';

it('renders correctly', () => {
  const composerGrid = renderer
    .create(
      <BrowserRouter>
        <ComposersGrid />
      </BrowserRouter>
    )
    .toJSON();
  expect(composerGrid).toMatchSnapshot();
});
