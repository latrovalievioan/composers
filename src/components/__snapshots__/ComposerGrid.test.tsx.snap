// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="search-bar"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-search fa-w-16 "
      data-icon="search"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 512 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <input
      onChange={[Function]}
      placeholder="Find by Name"
      type="text"
      value=""
    />
  </div>,
  <div
    className="dropdown-container"
  >
    <p>
      Sort By:
    </p>
    <div
      className="Dropdown-root"
    >
      <div
        aria-haspopup="listbox"
        className="Dropdown-control"
        onMouseDown={[Function]}
        onTouchEnd={[Function]}
      >
        <div
          className="Dropdown-placeholder is-selected"
        >
          name
        </div>
        <div
          className="Dropdown-arrow-wrapper"
        >
          <span
            className="Dropdown-arrow"
          />
        </div>
      </div>
    </div>
  </div>,
  <div
    className="composer-grid"
  />,
]
`;
